opent2t.p.thermostat/opent2t.p.thermostat.raml

```diff
@@ -0,0 +1,568 @@
+          "response": { "type": "string"}
+        }
+      }
+
+traits:
+  - interface-actuator:
+      queryParameters:
+        if:
+          enum: ["oic.if.a", "oic.if.baseline"]
+
+  - interface-sensor:
+      queryParameters:
+        if:
+          enum: ["oic.if.s", "oic.if.baseline"]
+
+resourceTypes:
+  - temperatureType:
+      get:
+        queryParameters:
+          units:
+            displayName: Units
+            enum: ["C","F","K"]
+        responses:
+          200:
+            body:
+              application/json:
+                schema: Temperature
+                example: |
+                  {
+                    "rt": ["oic.r.temperature"],
+                    "id": "unique_example_id",
+                    "temperature": 20.0,
+                    "units": "C",
+                    "range": [0.0,100.0]
+                  }
+      post:
+        body:
+          application/json:
+            schema: Temperature
+            example: |
+              {
+                "id": "unique_example_id",
+                "temperature":  18.0
+              }
+        responses:
+          200:
+             body:
+              application/json:
+                schema: Temperature
+                example: |
+                  {
+                    "id": "unique_example_id",
+                    "temperature": 18.0
+                  }
+  - sensorType:
+      get:
+        responses:
+          200:
+            body:
+              application/json:
+                schema: Sensor
+                example: |
+                  {
+                    "rt": ["oic.r.sensor"],
+                    "id": "unique_example_id",
+                    "value": true
+                  }
+
+  - modeType:
+      get:
+        responses:
+          200:
+            body:
+              application/json:
+                schema: Mode
+                example: |
+                  {
+                    "rt": ["oic.r.mode"],
+                    "id": "unique_example_id",
+                    "modes": <<modes>>,
+                    "supportedModes": <<supportedmodes>>
+                  }
+      post:
+        body:
+          application/json:
+            schema: ModeUpdate
+            example: |
+              {
+                "id":    "unique_example_id",
+                "modes": <<desiredmode>>
+              }
+        responses:
+          200:
+            body:
+              application/json:
+                schema: ModeUpdate
+                example: |
+                  {
+                    "id":    "unique_example_id",
+                    "modes": <<desiredmode>>
+                  }
+
+
+#
+# Device relative URIs
+#
+/devices/{deviceid}:
+  /ambientTemperature:
+    displayName: Ambient Temperature
+    is: [ interface-sensor ]
+    type: {temperatureType}
+
+  /targetTemperature:
+    displayName: Target Temperature
+    is: [ interface-actuator ]
+    type: {temperatureType}
```

Comment:
The 'type' field here references a type definition above. In this case, the temperatureType defines the get/post.

```raml
resourceTypes:
  - temperatureType:
      get:
        queryParameters:
          units:
            displayName: Units
            enum: ["C","F","K"]
        responses:
          200:
            body:
              application/json:
                schema: Temperature
                example: |
                  {
                    "rt": ["oic.r.temperature"],
                    "id": "unique_example_id",
                    "temperature": 20.0,
                    "units": "C",
                    "range": [0.0,100.0]
                  }
```

If you're asking in the context of validation, you should note that even if the temperatureType has both get and post, I will only implement post in addition to get if the resource is an 'interface-actuator' (essentially meaning read/write). If a resource is an 'interface-sensor', I only implement get.